File: backend/app/schemas/issue.py
from datetime import date
from pydantic import BaseModel

from app.schemas.article import Article, ArticleRead


class IssueSection(BaseModel):
    name: str
    description: str
    articles: list[Article]


class Issue(BaseModel):
    issue_date: date
    title: str
    tagline: str
    sections: list[IssueSection]


class IssueSectionRead(BaseModel):
    name: str
    description: str
    articles: list[ArticleRead]


class IssueRead(BaseModel):
    id: int
    issue_date: date
    title: str
    tagline: str
    status: str
    sections: list[IssueSectionRead]

File: frontend/src/types/issue.ts
export type Article = {
  id: number;
  title: string;
  url: string;
  source: string;
  published_at: string | null;
  summary: string | null;
  category: string | null;
};

export type IssueSection = {
  name: string;
  description: string;
  articles: Article[];
};

export type Issue = {
  id: number;
  issue_date: string;
  title: string;
  tagline: string;
  status: string;
  sections: IssueSection[];
};